source/Superluminal-RESTfulAPI-Tests/ExpiringResourceTest.class.st

```diff
@@ -8,11 +8,94 @@ Class {
 }
 
 { #category : #tests }
-ExpiringResourceTest >> testExpirationUsingMaxAge [
+ExpiringResourceTest >> currentDateAndTime [
+
+	^ DateAndTime now truncated
+]
+
+{ #category : #tests }
+ExpiringResourceTest >> testExpiredWhenExpiresSetToInvalidDate [
+
+	"A cache recipient MUST interpret invalid date formats, especially the
+   value 0, as representing a time in the past (i.e., 'already
+   expired').
+
+	See: https://tools.ietf.org/html/rfc7234#section-5.3"
 
 	| dateTime headers resource response |
 
-	dateTime := DateAndTime now.
+	dateTime := self currentDateAndTime.
+	headers := Dictionary new
+		at: 'Date' put: ( ZnUtils httpDate: dateTime );
+		at: 'Expires' put: 'not-a-date';
+		yourself.
+	response := ZnResponse noContent
+		headers: headers;
+		yourself.
+	resource := ExpiringResource for: 'here' controlledBy: response.
+	self
+		assert: resource contents equals: 'here';
+		assert: ( resource isExpiredAt: dateTime );
+		assert: ( resource isExpiredAt: dateTime + 1 second );
+		assert: ( resource isExpiredAt: dateTime - 1 second )
+]
+
+{ #category : #tests }
+ExpiringResourceTest >> testExpiredWhenExpiresSetToZero [
+
```

Comment:
There are many similar tests, the common code could be extracted, and maybe even leave only one test that calls many times with different collaborators the extracted code.